lib/shelley/bench/restore-bench.hs

```diff
@@ -596,6 +623,53 @@ benchmarksSeq _ w wid _wname benchname restoreTime = do
         , walletOverview
         }
 
+data BenchBaselineResults = BenchBaselineResults
+    { benchName :: Text
+    , restoreTime :: Time
+    } deriving (Show, Generic)
+
+instance Buildable BenchBaselineResults where
+    build = genericF
+
+instance ToJSON BenchBaselineResults where
+    toJSON = genericToJSON Aeson.defaultOptions
+
+{- HLINT ignore bench_baseline_restoration "Use camelCase" -}
+bench_baseline_restoration
+    :: forall (n :: NetworkDiscriminant) .
+        (
+         NetworkDiscriminantVal n
+        , HasNetworkId n
+        )
+    => Proxy n
+    -> Tracer IO (BenchmarkLog n)
+    -> Trace IO Text -- ^ For wallet tracing
+    -> CardanoNodeConn  -- ^ Socket path
+    -> NetworkParameters
+    -> NodeToClientVersionData
+    -> Text -- ^ Benchmark name (used for naming resulting files)
+    -> Bool -- ^ if to write to disk
+    -> Percentage -- ^ Target sync progress
+    -> IO SomeBenchmarkResults
+bench_baseline_restoration proxy tr wlTr socket np vData benchname traceToDisk  targetSync = do
+    putStrLn $ "*** " ++ T.unpack benchname
+    let networkId = networkIdVal proxy
+    withWalletLayerTracer benchname traceToDisk $ \progressTrace -> do
+        withNetworkLayer (trMessageText wlTr) networkId np socket vData sTol $ \nw -> do
+            wait <- newEmptyTMVarIO
+            synchronizer <- async $ chainSync nw nullTracer $ ChainFollower 
+                { readLocalTip  = pure []
+                , rollForward = \blocks _ntip -> do
+                    let (ntxs, NE.unzip -> (heights, slots)) = NE.unzip $ numberOfTransactionsInBlock <$>  blocks
+                    traceWith progressTrace $ Just $ NE.last heights
+                    seq (sum ntxs) $ atomically $ putTMVar wait $ NE.last slots
+                , rollBackward = pure
+                }
+            (_, restorationTime) <- bench "restoration" $ reportProgress nw tr targetSync (atomically $ takeTMVar wait)
+            cancel synchronizer
+            pure $ SomeBenchmarkResults (BenchBaselineResults benchname restorationTime)
```

Comment:
For clarity about the order of arguments, using the record field name for `BenchBaselineResults` is usuall  preferred over using them directly. Perhaps `NamedFieldPuns` can help here? Then you could name the variables `benchName` and `restoreTime` directly. — But in this case, I think the way you wrote it is fine, too.

Reply:
NamedFieldPuns applied!